### ISSUE TITLE
moveOverEventMessage.x is nil when tapped any node - AAChartViewDelegate

### ISSUE THREAD

**Issue by s-ahmad88** [AAChartModel/AAChartKit-Swift#246]
Hi, 

I am using Line chart in my app and I want to show a certain view when tapped on any Node(x,y) value in the graph. 
For some nodes(x,y) the values are fine for inside aaChartView() delegate, but for many the x value is nil and sometimes the y value is nil. I don't why its happening.

I only have to show that view for certain (x,y) values and if any of the value appears as nil then I cant verify it. 

Is there any solution to that ? Or anyone have any idea why x or y is appearing nil sometimes ? 
See the below image.

![Screen Shot 2020-08-31 at 10 47 31 am](https://user-images.githubusercontent.com/4697283/91673535-ee4ed200-eb77-11ea-9c15-4ace90bfbd98.png)

**Comment by AAChartModel** [comment]
Can you post your configuration code of AAChartModel or AAOptions?

**Comment by s-ahmad88** [comment]
`self.chartModel = AAChartModel()
            .chartType(AAChartType.line)
            .colorsTheme(self.currentSelectedGraphData.getSelectedColorTheme())//Colors theme
            .dataLabelsEnabled(true)
            .stacking(.none)
            .touchEventEnabled(true)
            .tooltipEnabled(false)
            .yAxisLabelsEnabled(true)
            .legendEnabled(false)
            .yAxisVisible(true)
            .yAxisMin(Float(self.currentSelectedGraphData.minYValue))
            .yAxisMax(Float(self.currentSelectedGraphData.maxYValue))
            .yAxisAllowDecimals(true)
            .xAxisVisible(false)
            .xAxisLabelsEnabled(false)
            .xAxisTickInterval(1.0)
            .animationType(.linear)
            .markerRadius(5)
            .markerSymbolStyle(.borderBlank)
            .markerSymbol(.circle)
            .series(self.currentSelectedGraphData.getSelectedGraphSeries())`

**Comment by s-ahmad88** [comment]
By the way, I have observed, this is even happening in the sample project of swift as well.

**Comment by AAChartModel** [comment]
I think may be it is a bug of Highcharts, However, Index values is always appears, so you can get x and y values by index 



```swift
x = moveOverEventMessage.index

y = aaSeriesElement.data[moveOverEventMessage.index]
```

**Comment by s-ahmad88** [comment]
Actually its not that simple. My X and Y axis are being generated at run time according to the data model and X axis could also be 0.1 as well. so Can't really reply on the above approach.

**Comment by AAChartModel** [comment]
I found the reason now. because JavaScript is dynamic language, so the types of x and y values are not very stable. sometimes it is `String` type, sometimes it is `Float` type, just like this as follow:

* `String` type y value


```js
user finger moved over!!!,get the move over event message: {
    category = "C++";
    index = 8;
    name = 2020;
    offset =     {
        plotX = "259.95833333333";
        plotY = "196.772604709689";
    };
    x = 8;
    y = "14.2";
}
```


* `Float` type y value

```js

user finger moved over!!!,get the move over event message: {
    category = C;
    index = 6;
    name = 2020;
    offset =     {
        plotX = "198.79166666667";
        plotY = "156.3052309904728";
    };
    x = 6;
    y = 17;
}
```

**Comment by AAChartModel** [comment]
It was fixed in this commit  b72b82b

**Comment by AAChartModel** [comment]
--------------------------------------------------------------------------------
 🎉  Congrats

 🚀  AAInfographics (5.0.3) successfully published
 📅  August 31st, 21:43
 🌎  https://cocoapods.org/pods/AAInfographics
 👍  Tell your friends!
--------------------------------------------------------------------------------
huanghunbieguannoMac:AAChartKit-Swift anan$
